src/utils/__tests__/retry.test.ts

```diff
@@ -0,0 +1,56 @@
+import { sleep } from '../../integration/testHelpers';
+import { retry } from '../retry';
+
+// Mock only `sleep`, keep other functions in testHelpers.ts as-is
+jest.mock('../testHelpers', () => ({
+  ...jest.requireActual('../testHelpers'),
+  sleep: jest.fn().mockResolvedValue(undefined), // Directly mock sleep here
+}));
+
+describe('retry function', () => {
+  beforeEach(() => {
+    jest.clearAllMocks(); // Clear mocks between tests
+  });
+
+  // Unit test
```

Comment:
Oops, I'll remove these

Reply:
Yeah I'd remove all the comments, I think?